openrtb_ext/request.go

```diff
@@ -197,141 +171,109 @@ type GranularityRange struct {
 	Increment float64 `json:"increment"`
 }
 
-// UnmarshalJSON : custom unmarshaller to handle legacy string granularites.
 func (pg *PriceGranularity) UnmarshalJSON(b []byte) error {
-	// We default to medium
-	if len(b) == 0 {
-		*pg = priceGranularityMed
-		return nil
-	}
-	// First check for legacy strings
-	var pgString string
-	err := json.Unmarshal(b, &pgString)
-	if err == nil {
-		*pg = PriceGranularityFromString(pgString)
-		if len(pg.Ranges) > 0 {
-			// Only exit if we matched something, else we try processing as custom granularity
-			// This way we error as expecting the new custom granularity standard.
+	// price granularity used to be a string referencing a predefined value, try to parse
+	// and map the legacy string before falling back to the modern custom model.
+	legacyID := ""
+	if err := json.Unmarshal(b, &legacyID); err == nil {
+		if legacyValue, ok := NewPriceGranularityFromLegacyID(legacyID); ok {
+			*pg = legacyValue
 			return nil
 		}
 	}
```

Comment:
Still need to keep custom unmarshal behavior for handling legacy id due to type difference (string vs object).